homework/12_B/19_Kristian/src/test/java/com/kris/circles/NotTouchTest.java

```diff
@@ -0,0 +1,52 @@
+package com.kris.circles;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.kris.circles.algorithms.CircleAlgorithms;
+
+class NotTouchTest {
+  Circle first;
+  Circle second;
+  CircleAlgorithms alg;
+
+  private final PrintStream standardOut = System.out;
+  private final ByteArrayOutputStream outputStreamCaptor = new ByteArrayOutputStream();
+
+  @BeforeEach
+  public void setUp() throws Exception {
+    System.setOut(new PrintStream(outputStreamCaptor));
+  }
+
+  @AfterEach
+  void tearDown() throws Exception {
+    System.setOut(standardOut);
+  }
+
+  @Test
+  void succeed() {
+    first = new Circle(24.8f, 13.123f, 2);
+    second = new Circle(-43.52f, -25.876f, 2);
+    alg = new CircleAlgorithms(first, second);
+
+    alg.run();
+    assertEquals("The circles DO NOT touch each other", outputStreamCaptor.toString().trim());
```

Comment:
This is not a particularly good way of doing tests. You are testing side effects and it is best for functions to have no sideeffects if possible. Intersection of circles is a easy example. It returns a nice result (an enum) no need to print anything, just test `calculate()` instead of `run()`.